packages/remix-server-runtime/routeModules.ts

```diff
@@ -76,9 +76,9 @@ export interface LoaderFunction {
  * `<meta>` tags for a route. These tags will be merged with (and take
  * precedence over) tags from parent routes.
  */
-export interface MetaFunction {
+export interface MetaFunction<Data = AppData> {
```

Comment:
I think we should make sure that it's an extension of `AppData`

```suggestion
export interface MetaFunction<Data extends AppData = AppData> {
```

Reply:
Do you think it would be useful? The AppData type is an alias of `any`
https://github.com/remix-run/remix/blob/31c33109b2cb68a615106dd5cf33b17d33ae10e1/packages/remix-server-runtime/data.ts#L12-L15
Maybe to be future proof if it's changed to something else?

Reply:
It's a good practice to extend the default type or some common thing imo.

Since it's `any` it doesn't really matter, but if it would ever be changed to something else (like an object or something else), it will make the `MetaFunction` generic a bit more type-safe.